docs/build/advanced-concepts/randomness.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Flow's Native Onchain Randomness
+sidebar_label: Native Randomness
+---
+
+# Native Onchain Randomness on Flow
+
+Flow enhances blockchain functionality by providing native onchain randomness, eliminating reliance on external oracles in Web3. This secure, decentralized feature empowers developers to build a variety of applications with truly unpredictable, transparent, and fair outcomes, achieved with greater efficiency.
+
+Flow’s onchain randomness delivers immediate random values within smart contracts, bypassing the latency and complexity of oracle integration. Developers can obtain verifiably random results with a single line of Cadence code, streamlining the development process and enhancing the performance of decentralized applications.
+
+## Use Cases of Onchain Randomness
+
+- **Gaming:** Integrates fairness and unpredictability into gameplay, enhancing user engagement without delays.
+- **NFTs:** Facilitates the creation of uniquely randomized traits in NFTs quickly, adding to their rarity and value.
+- **Lotteries & Draws:** Offers instant and verifiably fair random selection for lotteries, solidifying trust in real-time.
+- **DeFi Protocols:** Enables rapid and innovative random reward systems within decentralized finance.
+- **DAOs:** Assists in unbiased voting and task assignments through immediate randomness.
+- **Broad Applications:** Extends to any domain requiring impartial randomization, from asset distribution to security mechanisms, all with the added benefit of on-demand availability.
+
+## History - Distributed Random Beacon
+
+Within the Flow protocol, the heart of randomness generation lies in the "distributed random beacon." This module generates randomness that is distributed across the network while adhering to established safety and security standards. The output from the randomness beacon is a sequence of random bytes that are unpredictable and impartial.
+
+For over three years, the beacon has ensured protocol security by assigning verification nodes to oversee block computations. For those interested in a more detailed exploration of the randomness beacon and its inner workings, you can read [the technical deep dive on the Flow forum](https://forum.flow.com/t/secure-random-number-generator-for-flow-s-smart-contracts/5110).
+
```

Comment:
short section on history. Might need @tarakby's revisions 

```suggestion

### The History and Limitations of `unsafeRandom` (now deprecated)

Cadence has historically provided the `unsafeRandom` function to return a pseudo-random number. The stream of random numbers produced was potentially unsafe in the following two regards:

1. The sequence of random numbers is potentially predictable by transactions within the same block and by other smart contracts calling into your smart contract.
2. A transaction calling into your smart contract can potentially bias the sequence of random numbers which your smart contract internally generates. Currently, the block hash seeds `unsafeRandom`. Consensus nodes can *easily* bias the block hash and **influence the seed for `unsafeRandom`**.

⚠️ Note `unsafeRandom` will be deprecated in the next Cadence release. **Developers are advised to refrain from using `unsafeRandom.`**
```

Reply:
There are two other less important aspects but we don't necessarily need to mention them:
 3. the PRNG used did not provide high statistical quality.
 4. The sequence of random numbers is the same for each transaction of the same block
 
 I'm happy with the current state if we want to keep this brief.